app/controllers/pools_controller.rb

```diff
@@ -29,12 +29,52 @@ def edit
   end
 
   def update
-    @pool = Pool.find(params[:id])
-    new_params = pool_params
-    new_params[:active] = new_params.has_key?(:active) && new_params[:active] == "true"
-    unless @pool.update(new_params)
-      @errors = @pool.errors.full_messages
-      render status: :bad_request
+    if params[:id] && auth_admin?
+      # allow any valid change but concurrency unsafe
+      @pool = Pool.find(params[:id])
+
+      update_params = pool_params
+      update_params[:active] = update_params.has_key?(:active) &&
+                               update_params[:active] == "true"
+      if @pool.update(update_params)
```

Comment:
I figured this can raise errors if database returns an error for example for violating index. See #3 
Could you review #3?